shaft_element.py

```diff
@@ -1,478 +1,461 @@
-        n : int, optional
-            Element number (coincident with it's first node).
-            If not given, it will be set when the rotor is assembled
-            according to the element's position in the list supplied to
-            the rotor constructor.
-        axial_force : float
-            Axial force.
-        torque : float
-            Torque.
-        shear_effects : bool
-            Determine if shear effects are taken into account.
-            Default is False.
-        rotary_inertia : bool
-            Determine if rotary_inertia effects are taken into account.
-            Default is False.
-        gyroscopic : bool
-            Determine if gyroscopic effects are taken into account.
-            Default is False.
-        Returns
-        -------
-        elements: list
-            List with the 'ne' shaft elements.
-        Examples
-        --------
-        >>> # shaft material
-        >>> from ross.materials import steel
-        >>> # shaft inner and outer diameters
-        >>> si_d = 0
-        >>> so_d = 0.01585
-        >>> sec = ShaftElement.section(247.65e-3, 4, 0, 15.8e-3, steel)
-        >>> len(sec)
-        4
-        >>> sec[0].i_d
-        0
-        """
-
-        le = L / ne
-
-        elements = [
-            cls(le, si_d, so_d, material, n, shear_effects, rotary_inertia, gyroscopic)
-            for _ in range(ne)
-        ]
-
-        return elements
-
-    @classmethod
-    def load_from_xltrc(cls, file, sheet_name="Model"):
-        # TODO docstrings should be here not in the io module
-        geometry, materials = load_shaft_from_xltrc(file, sheet_name)
-        shaft = [
-            ShaftElement(
-                el.length,
-                el.id_Left,
-                el.od_Left,
-                materials[el.matnum],
-                n=el.elemnum - 1,
-            )
-            for i, el in geometry.iterrows()
-        ]
-
-        return shaft
-
-        #  TODO stiffness Matrix due to an axial load
-        #  TODO stiffness Matrix due to an axial torque
-        #  TODO add speed as an argument so that skew-symmetric stiffness matrix can be evaluated (default to None)
-        #  TODO skew-symmetric speed dependent contribution to element stiffness matrix from the internal damping
-        #  TODO add tappered element. Modify shaft element to accept i_d and o_d as a list with to entries.
+import numpy as np
+import pandas as pd
+import matplotlib.patches as mpatches
+from abc import ABC
+
+__all__ = [
+    "ShaftElement"
+]
+
+class Element(ABC):
+    """Element class."""
+
+    def __init__(self):
+        pass
+
+    def summary(self):
+        """A summary for the element.
+        A pandas series with the element properties as variables.
+        """
+        attributes = self.__dict__
+        attributes["type"] = self.__class__.__name__
+        return pd.Series(attributes)
+
+    # These are the abstract classes for mass, damping and stiffness matrices
+    def M(self):

```

Comment:
The decorator `@abstractmethod` is missing.
The purpose of the decorator is that, when inheriting from `element`, the subclass will have to implement the abstract method. If the method is not implemented a `TypeError` is raised.

Reply:
Just added the abstract methods.
These changes are shown in this commit [here](https://github.com/rodrigomoliveira1/ross/commit/189ddf53bdb508cd3dc2d61cca62c4182a6e2269)
I will address every commit link in each comment you've made, and the pull request will be done only after the last change.

Reply:
Great! The commits are already showing in this pull request, so there is no need to make another one.
Just one note on the commit title, it would be better if it was less generic. Instead of 'Update shaft_element.py' something as 'Add abstract methods'.

Reply:
Noted!